tools/c7n_gcp/c7n_gcp/resources/gke.py

```diff
@@ -79,6 +79,66 @@ def augment(self, resources):
                 r['labels'] = r['resourceLabels']
         return resources
 
+@KubernetesCluster.filter_registry.register('effective-firewall')
+class EffectiveFirewall(ValueFilter):
+    """Filters gke clusters  by their effective firewall rules.
+    See `getEffectiveFirewalls
+    https://cloud.google.com/workflows/docs/reference/googleapis/compute/v1/networks/getEffectiveFirewalls`_
+    for valid fields.
+
+    :example:
+
+    Filter all gke clusters that have a firewall rule that allows public
+    access
+
+    .. code-block:: yaml
+
+        policies:
+           - name: find-publicly-accessable-clusters
+             resource: gcp.gke-cluster
+             filters:
+             - type: effective-firewall
+               key: sourceRanges[]
+               op: contains
+               value: "0.0.0.0/0"
+    """
+
+    schema = type_schema('effective-firewall', rinherit=ValueFilter.schema)
+    permissions = ('compute.instances.getEffectiveFirewalls',)
+
+    def process_resource(self, client, p, network):
+        def get_port_ranges(ports):
+            port_ranges = []
+            for port in ports:
+                    if "-" in port:
+                        port_split = port.split("-")
+                        port_ranges.append({"beginPort": port_split[0], "endPort": port_split[1]})
+                    else:
+                        port_ranges.append({"beginPort": port, "endPort": port})
+            return port_ranges
+
+        firewalls = client.execute_command('getEffectiveFirewalls',
```

Comment:
the firewalls should ideally be cached on the resource as an annotationso multiple effective firewall filters in a policy don't need to refetch the firewalls.